jakarta-package-alignment-mappings/src/main/java/com/palantir/gradle/jakartapackagealignment/VersionMappings.java

```diff
@@ -235,6 +235,22 @@ private static Map<String, VersionMapping> createMappings() {
                         new MavenCoordinate("jakarta.xml.ws", "jakarta.xml.ws-api", "2.3.3"),
                         new MavenCoordinate("javax.xml.ws", "jaxws-api", "2.3.1")));
 
+        // glassfish hk2 repackages some components as an OSGi bundle
+        // we assume no OSGi usage in conjunction with this plugin, keeping the hk2 dependency
+        // around could lead to conflicting class names on the classpath
+        addMapping(
+                allMappings,
+                "org.glassfish.hk2.external:jakarta.inject",
+                new VersionMapping(
+                        new MavenCoordinate("org.glassfish.hk2.external", "jakarta.inject", "2.6.1"),
+                        new MavenCoordinate("javax.inject", "javax.inject", "1")));
+        addMapping(
+                allMappings,
+                "org.glassfish.hk2.external:javax.inject",
+                new VersionMapping(
+                        new MavenCoordinate("org.glassfish.hk2.external", "javax.inject", "2.4.0"),
+                        new MavenCoordinate("javax.inject", "javax.inject", "1")));
```

Comment:
Are there other libraries auto-resolved by the other plugin which aren't handled here?

Reply:
the other one would be the `el-api` things, but it actually goes in the opposite direction (`el-api` replaced with `org.glassfish:jakarta.el`).

I'm inclined to leave that one alone, it's not as straighforward since the `jakarta.el` implementation contains more than just the classes in `el-api`, whereas with `inject` it should be the same set of classes.